src/Compiler/Utilities/TypeHashing.fs

```diff
@@ -389,147 +389,190 @@ module StructuralUtilities =
 
     [<Struct; NoComparison; RequireQualifiedAccess>]
```

Comment:
Sorry to chime in, but I've accidentally noticed this pattern - struct DU with lots of cases, this usually leads to bad performance, because underlying memory is not shared and there is a big memory traffic because of it. If we are talking about optimization, then a struct tuple (enum + int), will work 6 times better memory-wise (if it's important) https://github.com/fsharp/fslang-suggestions/issues/699

Reply:
I think this is improved already, in this case sharplab shows the backing field is shared

Reply:
Its quite new, I added the optimization for the non-conflicting case of equal name and equal type (after erasures) during F# 9 period I believe.

The general case with overlapping memory for different types is still troublesome, indeed.